server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregatorTests.java

```diff
@@ -42,1236 +42,673 @@
-                        assertEquals(startDate.plusHours(i), bucket.getKey());
-                        assertEquals(1, bucket.getDocCount());
-                    }
-                });
-        testSearchAndReduceCase(new MatchAllDocsQuery(), dataset,
-                aggregation -> aggregation.setNumBuckets(300).field(DATE_FIELD),
-                histogram -> {
-                    List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
-                    assertEquals(200, buckets.size());
-                    for (int i = 0; i < 200; i++) {
-                        Histogram.Bucket bucket = buckets.get(i);
-                        assertEquals(startDate.plusHours(i * 3), bucket.getKey());
-                        assertEquals(3, bucket.getDocCount());
-                    }
-                });
-        testSearchAndReduceCase(new MatchAllDocsQuery(), dataset,
-                aggregation -> aggregation.setNumBuckets(100).field(DATE_FIELD),
-                histogram -> {
-                    List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
-                    assertEquals(50, buckets.size());
-                    for (int i = 0; i < 50; i++) {
-                        Histogram.Bucket bucket = buckets.get(i);
-                        assertEquals(startDate.plusHours(i * 12), bucket.getKey());
-                        assertEquals(12, bucket.getDocCount());
-                    }
-                });
-        testSearchAndReduceCase(new MatchAllDocsQuery(), dataset,
-                aggregation -> aggregation.setNumBuckets(30).field(DATE_FIELD),
-                histogram -> {
-                    List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
-                    assertEquals(25, buckets.size());
-                    for (int i = 0; i < 25; i++) {
-                        Histogram.Bucket bucket = buckets.get(i);
-                        assertEquals(startDate.plusDays(i), bucket.getKey());
-                        assertEquals(24, bucket.getDocCount());
-                    }
-                });
+        final Map<Integer, Integer> bucketsToExpectedDocCountMap = new HashMap<>();
+        bucketsToExpectedDocCountMap.put(seed, 1);
+        bucketsToExpectedDocCountMap.put(seed / 2, 3);
+        bucketsToExpectedDocCountMap.put(seed / 6, 12);
+        bucketsToExpectedDocCountMap.put(seed / 20, 24);
+        final Integer randomNumberOfBuckets = randomFrom(bucketsToExpectedDocCountMap.keySet());
+        testSearchAndReduceCase(DEFAULT_QUERY, dataset,
+            aggregation -> aggregation.setNumBuckets(randomNumberOfBuckets).field(DATE_FIELD),
+            histogram -> {
+                final List<? extends Histogram.Bucket> buckets = histogram.getBuckets();
+                final int expectedDocCount = bucketsToExpectedDocCountMap.get(randomNumberOfBuckets);
+                final int expectedSizeCeil = (seed + expectedDoc
```

Comment:
Removed this test as it felt kind of redundant. In this test case we first set the buckets so that data point is a key with doc count as 1, which is tested in testRandomHourInterval and testIntervalHour. Next, this sets the number of buckets to 6, a number such that keys are now 3 hour intervals, which is again tested by testRandomHourInterval. I did not see a special case being tested cases which the existing hour test cases could not test.

Reply:
I agree that part of this seems to be covered by the other randomized test, the case here seems to be somehow more carfuly crafted so I assumed it might have a special purpose. Maybe @colings86 who authored the initial commit of this test can elaborate more on this? If we want to keep it we should probably add a short note about the special purpose of this test.

Reply:
I'm +1 on removing this test.